schema/error.go

```diff
@@ -0,0 +1,22 @@
+package schema
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"go4.org/errorutil"
+)
+
+// WrapSyntaxError checks whether the given error is a json.Syntax error
+// and returns a wrapped error containing line/col information using the given reader.
+func WrapSyntaxError(r io.Reader, err error) error {
+	if serr, ok := err.(*json.SyntaxError); ok {
+		line, col, _ := errorutil.HighlightBytePosition(r, serr.Offset)
+
+		return errors.Wrapf(err, "line %d col %d", line, col)
```

Comment:
TODO(sur): instead of putting this information here, investigate whether we can unwrap wrapped `*json.SyntaxError` errs and produce a nice machine-readable output using the "standard" `file:line:col: msg` format.